packages/client/src/runtime/utils/getRuntime.ts

```diff
@@ -1,4 +1,68 @@
-import { detectRuntime, Runtime } from 'detect-runtime'
+// https://runtime-keys.proposal.wintercg.org/
+export type RuntimeName =
+  | 'workerd'
+  | 'deno'
+  | 'netlify'
+  | 'node'
+  | 'bun'
+  | 'edge-light'
+  | '' /* unknown */
+
+/**
+ * Indicates if running in Node.js or a Node.js compatible runtime.
+ *
+ * **Note:** When running code in Bun and Deno with Node.js compatibility mode, `isNode` flag will be also `true`, indicating running in a Node.js compatible runtime.
+ */
+const isNode = () => globalThis.process?.release?.name === 'node'
+
+/**
+ * Indicates if running in Bun runtime.
+ */
+const isBun = () => !!globalThis.Bun || !!globalThis.process?.versions?.bun
+
+/**
+ * Indicates if running in Deno runtime.
+ */
+const isDeno = () => !!globalThis.Deno
+
+/**
+ * Indicates if running in Netlify runtime.
+ */
+const isNetlify = () => typeof globalThis.Netlify === 'object'
+
+/**
+ * Indicates if running in EdgeLight (Vercel Edge) runtime.
+ */
+const isEdgeLight = () => typeof globalThis.EdgeRuntime === 'object'
+
+/**
+ * Indicates if running in Cloudflare Workers runtime.
+ * See: https://developers.cloudflare.com/workers/runtime-apis/web-standards/#navigatoruseragent
+ */
+const isWorkerd = () => globalThis.navigator?.userAgent === 'Cloudflare-Workers'
+  
+function detectRuntime(): RuntimeName {
+  // Note: we're currently not taking 'fastly' into account. Why?
+  const runtimeChecks = [
+    [isNetlify, 'netlify'],
+    [isEdgeLight, 'edge-light'],
+    [isWorkerd, 'workerd'],
+    [isDeno, 'deno'],
+    [isBun, 'bun'],
+    [isNode, 'node'],
+  ] as const
+
+  const detectedRuntime = runtimeChecks
+    // TODO: Transforming destructuring to the configured target environment ('chrome58', 'edge16', 'firefox57', 'safari11') is not supported yet,
+    // so we can't write the following code yet:
+    // ```
+    // .flatMap(([isCurrentRuntime, runtime]) => isCurrentRuntime() ? [runtime] : [])
```

Comment:
What why? Shouldn't esbuild do exactly the transformation you wrote yourself?

And why do we even target such old environments when we require Node.js 18+ or an equivalent modern runtime?

Reply:
I had a comment about this elsewhere: https://github.com/prisma/prisma/pull/26421#discussion_r1967539908